particle-iot/device-os
# Return relevant error from control request to enter listening mode

### Problem

In the case where listening mode is not allowed in the user firmware, the device-os returns `SYSTEM_ERROR_NONE` instead of the relevant error, which when propagated to the particle-cli results in a timeout.

### Solution

`processRequest()` which processes the control requests returns `SYSTEM_ERROR_NONE`. Added a blocking system call to return the relevant error from the `CTRL_REQUEST_START_LISTENING` control request. 

### Steps to Test

1. Run the following application on the module
2. Run `particle usb listen` on CLI and ensure "Not allowed" error is returned
```
test@Loaners-Macbook-Pro particle-usb (sc-92655/listening-mode-bug) $ particle usb listen
Not allowed
```


### Example App

```c
#include "application.h"
SerialLogHandler l(115200, LOG_LEVEL_ALL);
SYSTEM_MODE(MANUAL);
STARTUP(System.enableFeature(FEATURE_DISABLE_LISTENING_MODE));

void setup() {
}

void loop() {

}
```

### References

[SC-92655](https://app.shortcut.com/particle/story/92655/particle-cli-s-response-when-listening-mode-is-disabled-on-the-device-side)

---

### Completeness
